tools/generate_release_notes.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+# bash strict mode
+set -euo pipefail
+IFS=$'\n\t'
+
+display_usage() {
+	echo "Generate release notes for a release:"
+	echo -e "\nUsage:\n generate_release_notes.sh [previous_tag] [new_tag] \n"
+}
+
+# check whether user had supplied -h or --help . If yes display usage
+if [[ ( $# == "--help") ||  $# == "-h" ]]
+then
+	display_usage
+	exit 0
+fi
+
+# if less than two arguments supplied, display usage
+if [  $# -le 1 ] || [ $# -gt 2 ]
+then
+  echo -e "Incorrect number of arguments, expected 2 \n"
+	display_usage
+	exit 1
+fi
+
+PREVIOUS_TAG=$1
+NEW_TAG=$2
+
+# Verify that the tags are valid
+if ! git rev-parse "${PREVIOUS_TAG}" >/dev/null 2>&1; then
+  echo "Tag $PREVIOUS_TAG does not exist"
+  exit 1
+fi
+
+if ! git rev-parse "${NEW_TAG}" >/dev/null 2>&1; then
+  echo "Tag $NEW_TAG does not exist"
+  exit 1
+fi
+
+
+#######################################
+# Get the reference to upstream commits between the two tags
+# Globals:
+#   PREVIOUS_TAG
+#   NEW_TAG
+# Arguments:
+#   None
+# Outputs:
+#   A string with the format "upstream_commit1...upstream_commit2" that can be used as a reference to the upstream commits between the two tags
+#######################################
+function get_upstream_diff() {
+  local first_datadog_commit_name="Add options to control reporter interval / sampling frequency"
+
+  local previous_first_datadog_commit
+  previous_first_datadog_commit=$(git log "${PREVIOUS_TAG}" --grep="${first_datadog_commit_name}" --pretty=format:"%H" | tail -n 1)
+
+  local new_first_datadog_commit
+  new_first_datadog_commit=$(git log "${NEW_TAG}" --grep="${first_datadog_commit_name}" --pretty=format:"%H" | tail -n 1)
+
+  local previous_upstream
+  previous_upstream=$(git show -s --pretty=%P "${previous_first_datadog_commit}")
+
+  local new_upstream
+  new_upstream=$(git show -s --pretty=%P "${new_first_datadog_commit}"^1)
```

Comment:
```suggestion
  new_upstream=$(git show -s --pretty=%P "${new_first_datadog_commit}")
```
`%P` and `^1` might be redundant

Reply:
Missed this, great catch thanks! I'll update the changelog to fix this!